Microsoft.Bot.Builder.ApplicationInsights.Tests/BotTelemetryClientTests.cs

```diff
@@ -19,40 +20,49 @@ public void Construct()
             Assert.IsNotNull(client);
         }
 
+
+        [TestMethod]
+        public void TrackAvailabilityTest()
+        {
+            var telemetryClient = new TelemetryClient();
+            var client = new BotTelemetryClient(telemetryClient);
+
+            client.TrackAvailability("test", DateTimeOffset.Now, new TimeSpan(1000), "run location", true,
```

Comment:
Where is the assertion here?

Reply:
No assert, just straight call in case AppInsights interface has changed.

Reply:
Err, well... what if `BotTelemetryClient` suddenly just did nothing when `TrackAvailability` was called because someone made an oopsie?? This test would still pass because no exception was thrown. 

I'm arguing that these tests should be asserting that the implementation of `BotTelemetryClient` correctly invokes AI with the expected telemetry instance with all the correct details filled out. 

To do this you would instantiate a `TelemetryClient` instance with a `Mock<ITelemetryChannel>` associated with it that you can then `Verify` had its `Send` method called with the expected `ITelemetry` type and associated details from the parameters.

Reply:
Unfortunately, TelemetryClient is a sealed class with no interface.  If you find a good way of mocking it, let me know!

Reply:
@daveta You don't mock `TelemetryClient` itself, you mock an `ITelemetryChannel` that you hook to the `TelemetryClient` [as I said in my third paragraph here](https://github.com/Microsoft/botbuilder-dotnet/pull/1179/files#r235489993). 

[Here's a PR instead.](https://github.com/Microsoft/botbuilder-dotnet/pull/1185)!

Happy Thanksgiving! 🦃

Reply:
Ah, didn't see this - I added some tests in the latest checkin!  Thanks for the pointer, I used a similar method to test the TelemetryInitializer

Reply:
Thanks alot Drew!